.github/deployment/scripts/protocol-cli.rs

```diff
@@ -801,6 +890,166 @@ async fn update_chain_config(config: &DeploymentConfig, addresses_file: &str, ch
+        "arbitrum-sepolia" => NetworkInfo { chain_id: 421614, network: "testnet" },
+        "arbitrum" => NetworkInfo { chain_id: 42161, network: "mainnet" },
+        "ethereum-sepolia" => NetworkInfo { chain_id: 11155111, network: "testnet" },
+        "ethereum-mainnet" => NetworkInfo { chain_id: 1, network: "mainnet" },
+        "pecorino-signet" => NetworkInfo { chain_id: 14174, network: "testnet" },
+        "pecorino-host" => NetworkInfo { chain_id: 3151908, network: "testnet" },
+        _ => return None,
+    })
+}
+
+/// Encodes constructor arguments based on contract type
+fn encode_constructor_args(
+    constructor_type: &ConstructorType,
+    contract_name: &str,
+    env_vars: &HashMap<String, String>,
+) -> Result<String> {
+    match constructor_type {
+        ConstructorType::None | ConstructorType::Core => Ok(String::new()),
+        ConstructorType::Owner => {
+            let owner = env_vars.get("OWNER_ADDRESS")
+                .ok_or_else(|| anyhow!("OWNER_ADDRESS not found in environment"))?;
+            Ok(format!("000000000000000000000000{}", owner.trim_start_matches("0x")))
+        }
+        ConstructorType::Action => {
+            let fee_token_registry = env_vars.get("EXPECTED_FEE_TOKEN_REGISTRY_ADDRESS")
+                .ok_or_else(|| anyhow!("EXPECTED_FEE_TOKEN_REGISTRY_ADDRESS not found"))?;
+            let treasury = env_vars.get("EXPECTED_TREASURY_ADDRESS")
+                .ok_or_else(|| anyhow!("EXPECTED_TREASURY_ADDRESS not found"))?;
+            let gas_constant_var = format!("{}_GAS_CONSTANT", to_snake_case(contract_name));
+            let gas_constant = env_vars.get(&gas_constant_var)
+                .ok_or_else(|| anyhow!("Gas constant {} not found for {}", gas_constant_var, contract_name))?;
+
+            Ok(format!(
+                "000000000000000000000000{}000000000000000000000000{}{:0>64}",
+                fee_token_registry.trim_start_matches("0x"),
+                treasury.trim_start_matches("0x"),
+                format!("{:x}", gas_constant.parse::<u64>()?)
+            ))
+        }
+    }
+}
+
+/// Verifies contracts for networks based on a deployment config
+async fn verify_contracts(
+    config_file: &str,
+    env_dir: &str,
+    protocol_root: &str,
+    etherscan_api_key: Option<&str>,
+    verifier: &str,
+) -> Result<()> {
+    let config = load_deploy_config(config_file)?;
+    let chains = config.chains.as_ref().filter(|c| !c.is_empty()).ok_or_else(|| anyhow!("No chains in config"))?;
+
+    for chain in chains {
+        info(&format!("Verifying contracts for network: {}", chain));
+        let network_info = get_network_info(chain)
+            .ok_or_else(|| anyhow!("Unknown network: {}", chain))?;
+        let env_vars = load_chain_env_files(env_dir, chain, network_info.network)?;
+
+        for (key, value) in &env_vars {
+            env::set_var(key, value);
+        }
+
+        for contract_name in &config.contracts {
+            if contract_name == "Core" {
```

Comment:
nit: instead of 
if
else if
else

I would do:

```rust
if contract_name == "Core" { do stuff; continue }
if let Some(details) = ... {
 do stuff; continue
}
warn!
```

an even better approach is to have a helper method for that, then you'd have something like

```rust 
if ... { do_stuff; return }
if let .. {do_struff; return}
warn!(...)
```